nixos/modules/programs/zsh/zsh.nix

```diff
@@ -182,8 +147,36 @@ in
 
         . /etc/zinputrc
 
+        # history defaults
+        SAVEHIST=2000
+        HISTSIZE=2000
+        HISTFILE=$HOME/.zsh_history
+
+        setopt HIST_IGNORE_DUPS SHARE_HISTORY HIST_FCNTL_LOCK
+
+        # Tell zsh how to find installed completions
+        for p in ''${(z)NIX_PROFILES}; do
+          fpath+=($p/share/zsh/site-functions $p/share/zsh/$ZSH_VERSION/functions)
+        done
+
+        ${if cfg.enableCompletion then "autoload -U compinit && compinit" else ""}
+
+        ${optionalString (cfg.enableAutosuggestions)
+          "source ${pkgs.zsh-autosuggestions}/share/zsh-autosuggestions/zsh-autosuggestions.zsh"
+        }
+
+        ${zshAliases}
+
+        autoload -U promptinit && promptinit && prompt walters
+
+        ${cfge.interactiveShellInit}
+
+        HELPDIR="${pkgs.zsh}/share/zsh/$ZSH_VERSION/help"
```

Comment:
Same comment as the previous one; I only copy-pasted. It may be at the end of the file to prevent the user from setting `HELPDIR`. I can move it at the top if you wish so.